### PR TITLE
ref(ui): Refactor stream search bar into two components

### PR DESCRIPTION
**Pull request by billyvg**
This refactors the stream search bar into two components so that the base
component can be re-used. The stream search bar will provide supported
tags, default tags, and the API call to get tag values.

Required for https://github.com/getsentry/sentry/pull/10689

### CODE REVIEW

**Comment by billyvg** [comment]
Review the [most recent commit](https://github.com/getsentry/sentry/pull/10691/commits/dda656c62cbd0f773be06d33ad3834483e375906) for a better diff.

File: src/sentry/static/sentry/app/components/smartSearchBar.jsx [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+  };
+
+  /**
+     * Returns an array of query terms, including incomplete terms
+     *
+     * e.g. ["is:unassigned", "browser:\"Chrome 33.0\"", "assigned"]
+     */
+  static getQueryTerms = (query, cursor) => {
+    return query.slice(0, cursor).match(/\S+:"[^"]*"?|\S+/g);
+  };
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/smartSearchBar.jsx]: Is there a particular reason these are static class methods?

File: tests/js/spec/components/smartSearchBar.spec.jsx [hunk 2]
```diff
@@ -32,26 +32,30 @@ describe('removeSpace()', function() {
...
     };
 
-    urlTagValuesMock = MockApiClient.addMockResponse({
-      url: '/projects/123/456/tags/url/values/',
-      body: [],
-    });
+    // urlTagValuesMock = MockApiClient.addMockResponse({
+    // url: '/projects/123/456/tags/url/values/',
+    // body: [],
+    // });
```
Comment (EvanPurkhiser) [inline tests/js/spec/components/smartSearchBar.spec.jsx]: Can we remove this?

**Review by evanpurkhiser — APPROVED** [review]
Very nice work

File: src/sentry/static/sentry/app/components/smartSearchBar.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/smartSearchBar.jsx]: This was from previous code

**Pull request merged**